src/ansys/aedt/core/generic/settings.py

```diff
@@ -825,17 +832,26 @@ def filter_settings_with_raise(settings: dict, allowed_keys: List[str]):
                     raise KeyError("An environment variable key is not part of the allowed keys.")
                 self.aedt_environment_variables = settings
 
-    def writte_yaml_configuration(self, path: str):
+    def write_yaml_configuration(self, path: Path | str):
```

Comment:
Not sure if that 
```suggestion
    def write_yaml_configuration(self, path: Union[Path, str]):
```
The `|` syntax was introduced in python 3.10. Since we should be compatible, to some extent, with python 3.8, I wouldn't use `|` in settings.